scripts/add_component_skeleton.py

```diff
@@ -0,0 +1,196 @@
+rel_notes = (
+    "The release notes generated by this step are used when releasing a new "
+    "version of the project. The releases should document the changes between "
+    "the next version and the latest released."
+)
+
+dep_update = (
+    "Dependency updates are checked periodically. A project should codify the "
+    "the process for updating the working directory of the repository with "
+    "any changes to upstream dependencies. If changes are made, the job will "
+    "automatically create a new pull request."
+)
+
+std_pre = (
+    "pre is an optional executable for preparing the node prior to running "
+    "the build. This step is retried on failure given that it performs "
+    "preparatory actions and not the actual function of the job. Examples of "
+    "actions that should be taken here include things that would be performed "
+    "manually such as setting configuration options."
+)
+
+std_run = (
+    "run is a required executable that executes the purpose of the job. For "
+    "example, if the project deploys some software, this file would be used "
+    "to perform the deployment and then test the success of the operation."
+)
+
+std_post = (
+    "post is an optional executable that is the last stage to run. It is used "
+    "to perform actions that make use of the data generated. Such actions "
+    "include log aggregation and preparing the node to be used as an image."
+)
+
+std_env_vars = (
+    "Environment variables\n"
+    "RE_HOOK_ARTIFACT_DIR: The directory where job artifacts (log files, "
+    "configuration used, etc) should be written.\n"
+    "RE_HOOK_RESULT_DIR: The directory where test results should be written.\n"
+    "RE_JOB_NAME: The name of the job itself. For example "
+    "PM_repo-name-series-image-scenario-action\n"
+    "RE_JOB_IMAGE: The name of the 'image' as defined in the job.\n"
+    "RE_JOB_SCENARIO: The name of the 'scenario' as defined in the job.\n"
+    "RE_JOB_ACTION: The name of the 'action' as defined in the job.\n"
+    "RE_JOB_FLAVOR: The name of the 'flavor' as defined in the job.\n"
+    "RE_JOB_TRIGGER: The name of the 'trigger' that initiated the build, "
+    "examples include 'PULL', 'PUSH' and 'TIMER'.\n"
+    "RE_JOB_REPO_NAME: The project's GitHub repository which gets cloned in "
+    "the job."
+)
+
+rel_notes_env_vars = (
+    "Environment variables\n"
+    "RE_HOOK_VERSION: The version that is being released, e.g. 1.2.0.\n"
+    "RE_HOOK_PREVIOUS_VERSION: The version to compare against, e.g. 1.1.0.\n"
+    "RE_HOOK_RELEASE_NOTES: The file where release notes should be written. "
+    "The release notes should be formatted with GitHub-flavoured Markdown.\n"
+    "RE_HOOK_REPO_HTTP_URL: The URL for the repository being released e.g. "
+    "https://github.com/rcbops/rpc-openstack.git."
+)
+
+dep_update_env_vars = (
+    "Environment variables\n"
+    "READ_ONLY_TEST: When set to false a pull request is created to update "
+    "any dependencies that have changed."
```

Comment:
Do we need to expose this? Isn't this specified in the component's dep-update job itself and is used internally by dep_update.sh to determine if PRs, etc. should be created?

Reply:
I hadn't appreciated it was set in the JJB project definition. I'll remove it because it doesn't make sense here.